src/hooks/v2/contractReader/ETHReceivedFromTokens.ts

```diff
@@ -0,0 +1,80 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { BallotState } from 'models/ballot-state'
+import { useContext } from 'react'
+import { parseWad } from 'utils/formatNumber'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+
+import { MAX_REDEMPTION_RATE } from 'utils/v2/math'
+
+import useProjectReservedTokens from './ProjectReservedTokens'
+
+/**
+ * Returns the value in ETH that an amount of tokens can be redeemed for
```

Comment:
@johnnyd-eth I added some calucaltion notes here to help me make sense of it.

I also modified the code slightly do be based in MAX_REDEMPTION_RATE instead of percentages, meaning we don't have to convert redemption rate to percentage first.

It all appears to work the same, but wanted to make you aware incase something breaks.